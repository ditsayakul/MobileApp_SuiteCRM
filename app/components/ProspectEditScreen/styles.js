import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	
	container: {
		flex: 1,
		backgroundColor: "white",
		paddingTop: 50,
	},

	headerWrapper: {
		flex: 0.05,
		justifyContent: 'center',
		alignItems: 'center',
		padding: 5,
	},

	bodyWrapper: {
		flex: 0.6,
		padding: 10,
	},


	buttonWrapper: {
		flex: 0.05,
		flexDirection: 'row',
		justifyContent: 'space-around',
		alignItems: 'center',
		padding: 20,
	},

	input: {
		flex: 1,
	
	},

	scroll: {
		borderWidth: 2,
    	borderColor: '#CCCC',
	},

	icon: {
		height: 30,
		width: 30,
		backgroundColor: "#1F94B7",
	},

});