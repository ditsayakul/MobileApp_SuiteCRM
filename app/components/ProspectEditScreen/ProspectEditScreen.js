import React, { Component } from 'react';
import { Text, ScrollView, View, Button, TextInput, Alert, Image, ActivityIndicator } from 'react-native';
import { Toolbar, ThemeProvider, Icon, IconToggle } from 'react-native-material-ui';
import { default as MIcon } from 'react-native-vector-icons/MaterialCommunityIcons';

import { styles as defaultStyles } from '../../layout/styles.js'
import { styles, images } from './index.js'
import * as constants from '../../config/const.js'
import { restCall } from '../../lib/rest_api.js'

var DEBUG = false;

const uiTheme = {
    palette: {
        primaryColor: '#1F94B7',
    },
    toolbar: {
        container: {
            height: 50,
        },
    },
  };


export class ProspectEditScreen extends Component {

  static navigationOptions = {
        header: null,
  };

  constructor(props) {
    super(props);
    this.state = {
        isPushing: false,
        saveSucceed: undefined,
        hasModifications: false,
        last_name: null,
        first_name: null,
        title: null,
        service: null,
        account_name: null,
        phone_number: null,
        mobile_phone_number: null,
        website: null,
        email1: null,
        primary_address_street: null,
        primary_address_city: null,
        primary_address_postalcode: null,
        primary_address_country: null,
        description: null,
        deleted: 0,
        };
    this.handleCancel = this.handleCancel.bind(this);
    this.handleSave = this.handleSave.bind(this);
  }

  pushToServer(onSuccessMessage, onFailureMessage) {
    
    if(DEBUG){
      this.debugState();
    }
    var ip = this.props.navigation.state.params.ip;
    var session = this.props.navigation.state.params.session;

    var itemID = null;
    var updateListScreenFlatList = this.props.navigation.state.params.callback;

    if(this.props.navigation.state.params.item){
        itemID = this.props.navigation.state.params.item.name_value_list.id.value;
    }

    var nameValueList = [
                            {name:"name", value: ''},
                            {name:"id", value: (itemID) ? itemID : ''},
                            {name:"last_name", value: this.state.last_name},
                            {name:"first_name", value: this.state.first_name},
                            {name:"title", value: this.state.title},
                            {name:"department", value: this.state.service},
                            {name:"account_name", value: this.state.account_name},
                            {name:"phone_work", value: this.state.phone_number},
                            {name:"phone_mobile", value: this.state.mobile_phone_number},
                            {name:"website", value: this.state.website},
                            {name:"primary_address_street", value: this.state.primary_address_street},
                            {name:"primary_address_city", value: this.state.primary_address_city},
                            {name:"primary_address_postalcode", value: this.state.primary_address_postalcode},
                            {name:"primary_address_country", value: this.state.primary_address_country},
                            {name:"description", value: this.state.description},
                            {name:"email1", value: this.state.email1},
                            {name:"deleted", value: this.state.deleted},
                        ]
    var updatedData = {session: session,
                        module_name:"Leads",
                        name_value_list: nameValueList,
                      }

    var updatedDataJson = JSON.stringify(updatedData);                      
    
    this.setState({isPushing: true});

    var onSuccess = function(responseData){
        this.setState({isPushing: false});
        if(responseData.entry_list){

            console.log("(EditScreen received item) :");
            console.log(responseData.entry_list);
            this.setState({hasModifications: false});
            
            Alert.alert('Succès', onSuccessMessage,
                  [ 
                      {text: 'OK', onPress: () => {
                        updateListScreenFlatList(responseData.entry_list);
                        this.props.navigation.goBack();
                      }},
                  ],
                  {cancelable: false},
            )          
        } else {
            Alert.alert('Échec', onFailureMessage,
                  [ 
                      {text: 'OK', },
                  ]
            )
        }
    }

    var onFailure = function(error){
        this.setState({isPushing: false, error: true});
    }
  
    restCall("set_entry", updatedDataJson, ip, onSuccess.bind(this), onFailure.bind(this));
  
  }

  handleCancel(){

    var possibilities =  [
                {text: 'Non', },
                {text: 'Oui', onPress: () => this.props.navigation.goBack()},
            ];

    if(this.state.hasModifications){
        Alert.alert('Abandonner', "Êtes vous sur de vouloir abandonner vos modifications ?", possibilities);
    } else {
        this.props.navigation.goBack();
    }
  
  }

  handleSave(){
   
   if(!this.state.isPushing){
      this.state.deleted = 0;

      // Check if the input field values are corrects.
      if(this.checkInputFields()){
          this.pushToServer("Le prospect à bien été enregistré dans votre CRM",
          "Le prospect n'a pas pu être enregistré dans votre CRM");
      } 
    }
  }

  handleDelete(){

  var deleteProspect = function(){
      if(!this.state.isPushing){
          this.state.deleted = 1;
          this.pushToServer("Le prospect à bien été supprimé de votre CRM",
            "Le prospect n'a pas pu être supprimé de votre CRM");
      }
  }

  Alert.alert('Attention', "Êtes vous sûr de vouloir supprimer ce prospect ?",
                  [ 
                      {text: 'Non', },
                      {text: 'Oui', onPress: deleteProspect.bind(this)},
                  ]
            )
  }

  debugState(){
      console.log("this.state=");
      console.log(this.state);
  }

  checkInputFields(){
    // Better regex but not needed here. 
    //  /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;

    // Basic regex for email checking.
    var reg = /^[\w.-]+@[\w-]+\.[a-zA-Z]{2,4}$/;

    if(this.state.email1 && !reg.test(this.state.email1)){
       Alert.alert('Erreur', 'Format de l\'email incorrect',
                  [ 
                      {text: 'OK'},
                  ]
        )
        return false;
    }

    if(!this.state.last_name){
       Alert.alert('Erreur', 'Le champ \'Nom\' est vide',
                  [ 
                      {text: 'OK'},
                  ]
        )
        return false;
    }
    return true;
  }

  componentWillMount(){
      var item = this.props.navigation.state.params.item;

      if(item){
        item = item.name_value_list;
          for(key in item){
              if(item[key].value){
                  if(DEBUG){
                      console.log("Key retireved: "+ key + "   value =  "+ item[key].value);
                  }
                  this.setState({[key]: item[key].value});
              }
          }
      }
  }

  updateData(key, value){
      key.value = value;
      this.setState({[key]: value, hasModifications: true});
  }

  render() {
    	return (
        <ThemeProvider uiTheme={uiTheme}>
        		<View style={styles.container}>
        					
                <Toolbar
                    ref={toolbarComponent => this.toolbar = toolbarComponent}
                    key="toolbar"
                    leftElement={<IconToggle name="navigate-before" color="white" onPress={this.handleCancel} disabled={this.state.isPushing}/>}
                    rightElement={<IconToggle name="save" color="white" onPress={this.handleSave} disabled={this.state.isPushing}/>}
                    centerElement="Edition du prospect"
                />

        				<View style={styles.headerWrapper}>
                    {/*Only display the prospect ID field if it exists*/}
                    {this.props.navigation.state.params.item &&
        					       <Text>Prospect ID: {this.props.navigation.state.params.item.name_value_list.id.value} </Text>
        				    }
                </View>

        				<View style={styles.bodyWrapper}>
                  {this.state.isPushing &&
                  <ActivityIndicator style={styles.headerWrapper} size="large" /> ||

        					   <ScrollView style={styles.scroll}>
        						      <InputLabelRow icon='account-circle' value = {this.state.last_name} onChangeText = {(text) => this.updateData("last_name" ,text)} placeholder='Nom'/>
        						      <InputLabelRow icon={null} value = {this.state.first_name} onChangeText = {(text) => this.updateData("first_name", text)} placeholder='Prénom'/>
        						      <InputLabelRow icon={null} value = {this.state.title} onChangeText = {(text) => this.updateData("title",text)} placeholder='Fonction'/>
        						      <InputLabelRow icon={null} value = {this.state.department} onChangeText = {(text) => this.updateData("department", text)} placeholder='Service'/>
        						      <InputLabelRow icon={null} value = {this.state.account_name} onChangeText = {(text) => this.updateData("account_name", text)} placeholder='Nom de compte'/>
        					      	<InputLabelRow micon='phone-classic' value = {this.state.phone_work} onChangeText = {(text) => this.updateData("phone_work", text)} placeholder='Téléphone fixe' keyboardType='phone-pad'/>
        					     	  <InputLabelRow micon='cellphone' value = {this.state.phone_mobile} onChangeText = {(text) => this.updateData("phone_mobile", text)} placeholder='Téléphone mobile' keyboardType='phone-pad'/>
                          <InputLabelRow icon='mail-outline' value = {this.state.email1} onChangeText = {(text) => this.updateData("email1", text)} placeholder='E-mail'/>
        						      <InputLabelRow micon='web' value = {this.state.website} onChangeText = {(text) => this.updateData("website", text)} placeholder='Site web' keyboardType='url' />
                          <InputLabelRow icon='edit-location' value = {this.state.primary_address_country} onChangeText = {(text) => this.updateData("primary_address_country", text)} placeholder='Pays'/>
                          <InputLabelRow icon={null} value = {this.state.primary_address_city} onChangeText = {(text) => this.updateData("primary_address_city", text)} placeholder='Ville'/>
                          <InputLabelRow icon={null} value = {this.state.primary_address_street} onChangeText = {(text) => this.updateData("primary_address_street", text)} placeholder='Rue'/>
                          <InputLabelRow icon={null} value = {this.state.primary_address_postalcode} onChangeText = {(text) => this.updateData("primary_address_postalcode", text)} placeholder='Code Postal' keyboardType='numeric'/>
        						      <InputLabelRow icon='description' multiline={true} value = {this.state.description} onChangeText = {(text) => this.updateData("description", text)} placeholder='Description'/>
        					   </ScrollView>
                  }
        				</View>
        			
                {/*Only display the delete button when the screen is accessed from an existing prospect*/}
                {this.props.navigation.state.params.item &&

        				    <View style={styles.buttonWrapper}>
        						    <Button
                  				  onPress={() => this.handleDelete()}
                 					  title= "Delete"
                  				  color="red"
                  				  accessibilityLabel="Delete the current prospect"
                            disabled={this.state.isPushing}
                		    />
        				    </View>
        			  }
        		</View>
        </ThemeProvider>
    		);
  }

}

var InputLabelRow = React.createClass({

   	render() {
    	return (

    		<View style={{
    			flex: 1,
    			flexDirection: 'row',
				  alignItems: 'center',
			  }}>
            <View style={{
              height: 30,
              width: 30,
            }}>
                {this.props.icon &&
      				      <Icon name={this.props.icon} size={30}/>
                }

                {this.props.micon &&
                    <MIcon name={this.props.micon} size={30}/>
                }
          </View>
				  <TextInput 
              multiline={this.props.multiline}
              style={{flex: 1}}
              value = {this.props.value}
              onChangeText = {this.props.onChangeText}
              placeholder={this.props.placeholder}
              keyboardType = {this.props.keyboardType ? this.props.keyboardType : 'default'}
          />
    		</View>

    	);
   	}
  });