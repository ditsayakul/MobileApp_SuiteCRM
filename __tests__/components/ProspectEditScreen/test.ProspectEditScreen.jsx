import 'react-native';
import React from 'react';
import { ProspectEditScreen } from '../../../app/components/ProspectEditScreen/ProspectEditScreen'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


/* Mock for the default props of the component. */
var arg = {
	state: {
		params:{
			item:null,
		}
	}
}

var prospectMock = require('./prospectMock.json');

afterEach(() => {
  // Reload the default configuration after each test.
  arg = {
		state: {
			params:{
				item:null,
			}
		}
	}
});



describe('ProspectEditScreen', () => {
    it('renders without crashing', () => {
    	const tree = renderer.create(
    		<ProspectEditScreen navigation={arg}/>
    	);
    });
    it('renders without prospect infos', () => {
    	const tree = renderer.create(
    		<ProspectEditScreen navigation={arg}/>
    	);
    	const instance = tree.getInstance();
    	// Because the arg passed to the component doesn't have any item,
    	// we expect the state[key] of the instance to be set to null.
    	for (key in prospectMock.name_value_list) {
				expect(instance.state[key]).toBeNull();
		}
 	});
    it('renders with prospect info', () => {
    	arg.state.params.item = prospectMock;
    	const tree = renderer.create(
    		<ProspectEditScreen navigation={arg}/>
    	);

    	const instance = tree.getInstance();
		for (key in prospectMock.name_value_list) {
			
			// If the value is set then we expect to find it in the component state.
			if(prospectMock.name_value_list[key].value){
				expect(instance.state[key]).toBe(prospectMock.name_value_list[key].value);
			}
			// Otherwise, the value should be set to null.
			else {
				expect(instance.state[key]).toBeNull();
			}
		}
    });
    it('is editable by default on creation of prospect', () => {
    	const tree = renderer.create(
    		<ProspectEditScreen navigation={arg}/>
    	);
    	const instance = tree.getInstance();
		expect(instance.state.isEditable).toBeTruthy();
 	});
 	it('is not editable by default on consultation of prospect', () => {
    	arg.state.params.item = prospectMock;
    	const tree = renderer.create(
    		<ProspectEditScreen navigation={arg}/>
    	);
    	const instance = tree.getInstance();
    	expect(instance.state.isEditable).toBeFalsy();
 	});
});