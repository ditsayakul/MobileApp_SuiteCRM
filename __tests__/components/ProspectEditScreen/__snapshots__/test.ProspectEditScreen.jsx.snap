// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F94B7",
        "flexDirection": "row",
        "height": 50,
        "overflow": "hidden",
        "paddingHorizontal": 4,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#1F94B7",
            "borderRadius": 0,
            "height": 0,
            "position": "absolute",
            "right": undefined,
            "top": -0,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 0,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "height": 0,
            "left": undefined,
            "position": "absolute",
            "top": -0,
            "transform": Array [
              Object {
                "scale": 0.01,
              },
            ],
            "width": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={undefined}
        testID={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderRadius": 21.6,
              "height": 43.2,
              "left": 2.3999999999999986,
              "opacity": 0.16,
              "position": "absolute",
              "top": 2.3999999999999986,
              "transform": Array [
                Object {
                  "scale": 0.01,
                },
              ],
              "width": 43.2,
              "zIndex": 1,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 48,
                "justifyContent": "center",
                "width": 48,
              },
              Object {
                "height": 48,
                "width": 48,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "marginLeft": 20,
          "opacity": 1,
        }
      }
      testID={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0)",
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Création
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flexDirection": "row",
          },
          false,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={undefined}
          testID={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.38)",
                "borderRadius": 21.6,
                "height": 43.2,
                "left": 2.3999999999999986,
                "opacity": 0.16,
                "position": "absolute",
                "top": 2.3999999999999986,
                "transform": Array [
                  Object {
                    "scale": 0.01,
                  },
                ],
                "width": 43.2,
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 48,
                  "justifyContent": "center",
                  "width": 48,
                },
                Object {
                  "height": 48,
                  "width": 48,
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.38)",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={undefined}
          testID={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderRadius": 21.6,
                "height": 43.2,
                "left": 2.3999999999999986,
                "opacity": 0.16,
                "position": "absolute",
                "top": 2.3999999999999986,
                "transform": Array [
                  Object {
                    "scale": 0.01,
                  },
                ],
                "width": 43.2,
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 48,
                  "justifyContent": "center",
                  "width": 48,
                },
                Object {
                  "height": 48,
                  "width": 48,
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.05,
        "justifyContent": "center",
        "padding": 5,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 0.6,
        "padding": 10,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#CCCC",
          "borderTopWidth": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Nom"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Prénom"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Fonction"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Service"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Nom de compte"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="phone-pad"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Téléphone fixe"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="phone-pad"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Téléphone mobile"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="E-mail"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="url"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Site web"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Pays"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Ville"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Rue"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <TextInput
            editable={true}
            keyboardType="numeric"
            multiline={undefined}
            onChangeText={[Function]}
            placeholder="Code Postal"
            style={
              Object {
                "flex": 1,
                "height": 50,
              }
            }
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 100,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="default"
            multiline={true}
            onChangeText={[Function]}
            placeholder="Description"
            style={
              Object {
                "flex": 1,
                "height": 100,
              }
            }
            value={null}
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
